### PR TITLE
2.3.0 release

### PR DESCRIPTION
**Pull request by mnive93**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,7 +50,7 @@ You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
 	  }
   }
   dependencies {
-      implementation('co.hyperverge:hypersnapsdk:2.2.0@aar', {
+      implementation('co.hyperverge:hypersnapsdk:1.0.8@aar', {
```
Comment (sanchit-mehta) [inline README.md]: How has version number reduced?

File: README.md [hunk 2]
```diff
@@ -78,7 +72,7 @@ You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
   }
   ```
   where **aws_access_key** and **aws_secret_pass** will be given by HyperVerge
-
+
```
Comment (sanchit-mehta) [inline README.md]: Trailing spaces

File: README.md [hunk 3]
```diff
@@ -87,61 +81,75 @@ You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
     Kindly note that for android v23 (Marshmallow) and above, you need to handle the runtime permissions inside your app.
 
 ### 3. Initializing the SDK
-- Add the following line to your Application class (the class which extends android.app.Application) for initializing our Library. This must be run only once. Check [this](https://guides.codepath.com/android/Understanding-the-Android-Application-Class) link if you are unsure of what an Application class is.
+- Add the following line to your Application class (the class which extends android.app.Application) for initializing our Library. This must be run only once. Check [this](https://guides.codepath.com/android/Understanding-the-Android-Application-Class) link if you are unsure of what an Application class is.
```
Comment (sanchit-mehta) [inline README.md]: Trailing space

File: README.md [hunk 4]
```diff
@@ -87,61 +81,75 @@ You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
...
-	- product: This is an enum, `HyperSnapParams.Product` with two values - `faceID` annd `faceIAM`. Right now, only `faceID` is supported.
->**Note**: This step is required only if liveness is used.
+	- product: This is an enum, `HyperSnapParams.Product` with two values - `faceID` annd `faceIAM`. 
 
 ### 4. Launching the Activities
 
 #### For Document Capture
  For capturing documents (based on Aspect Ratio), following method should be called:
-
+
```
Comment (sanchit-mehta) [inline README.md]: Trailing space

File: README.md [hunk 5]
```diff
@@ -161,25 +169,31 @@ CaptureCompletionHandler is an interface whose object needs to be passed with st
...
 
 #### Liveness in Face Capture:
-The SDK has two liveness detection methods. Texture liveness and Gesture Liveness. This can be set using the `livenessMode` parameter in the `start` method of `HVFaceActivity` discussed earlier.
+The SDK has two liveness detection methods. Texture liveness and Gesture Liveness. This can be set using the `livenessMode` parameter in the `setLivenessMode` method of `HVFaceConfig` discussed earlier.
 
-Here, `livenessMode` is of type `HypeSnapParams.LivenessMode`, an enum with 3 values:
+Here, `livenessMode` is of type `HVFaceConfig.LivenessMode`, an enum with 3 values:
 
-**.none**: No liveness test is performed. The selfie that is captured is simply returned. If successful, the result JSON in the CaptureCompletionHandler has one key-value pair.
+**.none**: No liveness test is performed. The selfie that is captured is simply returned. If successful, the result JSON in the CaptureCompletionHandler has one key-value pair.
```
Comment (sanchit-mehta) [inline README.md]: Trailing space

File: .idea/codeStyles/Project.xml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (amnosuperman) [inline .idea/codeStyles/Project.xml]: Should .idea folder be pushed?

File: CHANGELOG.md [hunk 7]
```diff
@@ -29,3 +29,10 @@
 - Modified localisation support to ease integration with react native
 - Performance improvements
 - Minor bug fixes
+
+### Version 2.3.0
+-  Added optional Instruction Pages for Face and Document Capture
+-  Added Review Page for Document capture
+-  Added provision for style customisation for all prominent labels and buttons
+-  Added optional 'clientID' field to liveness calls
+-  Added customisation options for strings and colors used in the SDK
```
Comment (amnosuperman) [inline CHANGELOG.md]: Swap order of points 4 and 5

File: CHANGELOG.md [hunk 8]
```diff
@@ -29,3 +29,10 @@
 - Modified localisation support to ease integration with react native
```
Comment (amnosuperman) [inline CHANGELOG.md]: Reverse order of releases in the changelog. The most recent release should be at the top and the oldest should be at the bottom

File: README.md [hunk 9]
```diff
@@ -1,25 +1,20 @@
-
-## HyperSnapSDK Documentation for Android
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 10]
```diff
@@ -1,25 +1,20 @@
...
+## HyperSnapSDK Documentation
 
 ## Overview
 
 HyperSnapSDK is HyperVerge's documents + face capture SDK that captures images at a resolution appropriate for our proprietary Deep Learning OCR and Face Recognition Engines.
 
-The framework provides a liveness feature that uses our advanced AI Engines to differentiate between a real user capturing his/her selfie from a photo/video recording.
+The framework provides a liveness feature that uses our advanced AI Engines to tell if a captured image is that of a real person or a photograph.
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 11]
```diff
@@ -1,25 +1,20 @@
...
 
 ### Requirements
 - Gradle Version: 4.1 (Recommended)
 - Tested with Gradle Plugin for Android Studio - version 3.0.1
 - minSdkVersion 19
 - targetSdkVersion 26
 
-### ChangeLog
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 12]
```diff
@@ -1,25 +1,20 @@
...
-You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
-
 ## Table Of Contents
 - [Overview](#overview)
 	- [Requirements](#requirements)
-	- [ChangeLog](#changelog)
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md (hunk 1 above)
Comment (amnosuperman) [inline README.md]: Wrong version number

File: README.md [hunk 13]
```diff
@@ -87,61 +81,75 @@ You can find the ChangeLog in the [CHANGELOG.md](CHANGELOG.md) file
...
+- Add the following line to your Application class (the class which extends android.app.Application) for initializing our Library. This must be run only once. Check [this](https://guides.codepath.com/android/Understanding-the-Android-Application-Class) link if you are unsure of what an Application class is. 
   ```java
   HyperSnapSDK.init(context, APP_ID, APP_KEY, region, product);
   ```
 	Where,
 	- appId, appKey are given by HyperVerge
 	- region: This is an enum, `HypeSnapParams.Region` with three values - `AsiaPacific`, `India` and `UnitedStates`.
-	- product: This is an enum, `HyperSnapParams.Product` with two values - `faceID` annd `faceIAM`. Right now, only `faceID` is supported.
->**Note**: This step is required only if liveness is used.
+	- product: This is an enum, `HyperSnapParams.Product` with two values - `faceID` annd `faceIAM`.
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 14]
```diff
@@ -161,25 +169,31 @@ CaptureCompletionHandler is an interface whose object needs to be passed with st
...
-**.none**: No liveness test is performed. The selfie that is captured is simply returned. If successful, the result JSON in the CaptureCompletionHandler has one key-value pair.
+**.none**: No liveness test is performed. The selfie that is captured is simply returned. If successful, the result JSON in the CaptureCompletionHandler has one key-value pair. 
 - `imageUri` : local path of the image captured
 
 **.textureLiveness** : Texture liveness test is performed on the selfie captured.  If successful, a result JSON with the following key-value pairs is returned in the CaptureCompletionHandler
 
 - `imageUri` : String. Local path of the image captured <br/>
 - `live`: String with values 'yes'/'no'. Tells whether the selfie is live or not.
-- `liveness-score`: Float with values between 0 and 1. The confidence score for the liveness prediction.
-- `to-be-reviewed`: String with values 'yes'/'no'. Yes indicates that it is flagged for manual review.
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 15]
```diff
@@ -161,25 +169,31 @@ CaptureCompletionHandler is an interface whose object needs to be passed with st
...
+- `liveness-score`: Float with values between 0 and 1.
+- The confidence score for the liveness prediction.
+- `to-be-reviewed`: String with values 'yes'/'no'. Yes indicates that it flagged for manual review.
 
-**.textureAndGestureLiveness**: In this mode, based on the results of the texture Liveness call, the user might be asked to do a series of gestures to confirm liveness. This mode is currently in beta. It is highly recommended to not use it in production.
+**.textureAndGestureLiveness**: In this mode, based on the results of the texture Liveness call, the user might be asked to do a series of gestures to confirm liveness. The user performing the gestures is arbitrarily matched with the selfie captured. If  one or more of these matches fail, a 'faceMatch' error is returned (refer to 'Error Codes' section).
```
Comment (amnosuperman) [inline README.md]: Retain original

File: README.md [hunk 16]
```diff
@@ -196,17 +210,127 @@ Descriptions of the error codes returned in the CaptureCompletionHandler are giv
...
-**strings.xml**:
-
-      <string name="document_screen_title_text">Document Scanner</string>
-      <string name="face_screen_title_text">Face Scanner</string>
-      <string name="placeFaceInCircle">Place your face within circle</string>
-      <string name="captureNow"> Capture Now </string>
-      <string name="moveAway"> Move away from the phone</string>
-      <string name="activityText">Processing</string>
-
+- Some text fields, element colors, font styles, and button icons are customizable so that the look and feel of the components inside SDK can be altered to match the look and field of the app using the SDK. **Kindly note that this step is optional**. Below is the list of items that are customizable grouped by the resource file/type where the customized value/file(s) should be placed in.
```
Comment (amnosuperman) [inline README.md]: Typo "look and field" of the app

File: README.md [hunk 17]
```diff
@@ -196,17 +210,127 @@ Descriptions of the error codes returned in the CaptureCompletionHandler are giv
...
+             <string name="docInstructions2">Do not place outside</string>
+             <string name="docInstructions3">Avoid glare</string>
+        ```
+      
+       - **Document capture review page strings**:
+           ```xml
+              <string name="docReviewRetakeButton">Retake</string>
+              <string name="docReviewContinueButton">Use This Photo</string>
+              <string name="docReviewTitle">Review Your Photo</string>
+              <string name="docReviewDescription">Is your document fully visible, glare free and not blurred ?</string>
```
Comment (amnosuperman) [inline README.md]: Remove space before question mark

File: sample/app/src/main/java/co/hyperverge/hypersnapsample/MainActivity.java [hunk 18]
```diff
@@ -206,32 +211,45 @@ public void run() {
...
+            startAppropriateDocumentActivity(docConfig);
         }
         if (view.getId() == R.id.tv_card) {
-            selectedDocument = HyperSnapParams.Document.CARD;
-            selectedDocument.setTopText("Card Front Side");
-            selectedDocument.setBottomText("Place your Card in the box");
-            startAppropriateDocumentActivity(selectedDocument);
+            selectedDocument = HVDocConfig.Document.CARD;
+            docConfig.setCaptureScreenTitleText("Docs Capture");
+            selectedDocument.setCapturePageInstructionText("Make sure your document is without any. Glare and is fully inside");
```
Comment (amnosuperman) [inline sample/app/src/main/java/co/hyperverge/hypersnapsample/MainActivity.java]: Typo - Make sure your document is without any. Glare and is fully inside

File: README.md (hunk 14 above)
Comment (mnive93) [reply README.md]: Hey I have not removed this and it's present in the documentation.

**Pull request merged**